src/Framework/Framework/Diagnostics/DiagnosticsStartupTracer.cs

```diff
@@ -30,34 +32,27 @@ public void TraceEvent(string eventName)
             }
 
             var eventTiming = CreateEventTiming(eventName);
-            bool reportLateEvent;
-            lock (locker)
-            {
-                events.Add(eventTiming);
-                reportLateEvent = startupCompleted;
-            }
+            ConcurrencyUtils.CasChange(ref events, l => l.Add(eventTiming));
+            var reportLateEvent = startupCompleted > 0;
 
             if (reportLateEvent)
```

Comment:
Technically, this is a race. Although, I am not sure how does this part of the framework works. It could be benign.

* T1: on L36 evaluates `reportLateEvent`
* T2: on L47 increments `startupCompleted`
* T1 then either executes this condition or not - depending on how the previous operations are scheduled.

Reply:
You are right, should be gone now (there is only one variable which we need to synchronize)